test/e2e/aadpodidentity_test.go

```diff
@@ -231,7 +231,10 @@ var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
 			identityValidatorName := fmt.Sprintf("identity-validator-%d", i)
 
 			setUpIdentityAndDeployment(keyvaultIdentity, fmt.Sprintf("%d", i))
-			time.Sleep(5 * time.Second)
+
```

Comment:
Removed the sleep, so the test can actually proceed once the assigned identity exists.